marklogic-client-api/src/main/java/com/marklogic/client/impl/HTTPSamlAuthInterceptor.java

```diff
@@ -29,96 +29,98 @@
 import okhttp3.Request;
 import okhttp3.Response;
 
-public class HTTPSamlAuthInterceptor implements Interceptor, Runnable {
-
-	  private String authorizationTokenValue;
-	  private AuthorizerCallback authorizer;
-	  private static ExpiringSAMLAuth expiringSAMLAuth;
-	  private static long threshold;
-	  private static RenewerCallback renewer;
-	  static AtomicBoolean isCallbackExecuting = new AtomicBoolean(false);
-
-	  HTTPSamlAuthInterceptor(String authToken) {
-	    this.authorizationTokenValue = authToken;
-	  }
-	  HTTPSamlAuthInterceptor(AuthorizerCallback authorizer) {
-		this.authorizer = authorizer;
-	  }
-	  public HTTPSamlAuthInterceptor(ExpiringSAMLAuth authorization, RenewerCallback renew) {
-	    expiringSAMLAuth = authorization;
+public class HTTPSamlAuthInterceptor implements Interceptor { 
+                                                                       
+    private String authorizationTokenValue;
+    private AuthorizerCallback authorizer;
+    private ExpiringSAMLAuth expiringSAMLAuth;
+    private long threshold;
+    private RenewerCallback renewer;
+    private AtomicBoolean isCallbackExecuting = new AtomicBoolean(false);
```

Comment:
Given the belated realization that the compareAndSet() can't be used for the authorizer (see below), maybe isCallbackExecuting should be initialized only in the constructor that takes the renewer.